classes/fields/text.php

```diff
@@ -165,7 +180,7 @@ public function validate( $value, $name = null, $options = null, $fields = null,
 		if ( is_array( $check ) ) {
 			$errors = $check;
 		} else {
-			if ( 0 < strlen( $value ) && '' === $check ) {
+			if ( $this->is_empty( $check ) ) {
```

Comment:
This could simplify things, we only care about strings here and it's either empty or not.

```suggestion
			if ( '' !== $value && '' === $check ) {
```